src/components/navigation/Navigation.tsx

```diff
@@ -35,10 +37,14 @@ export const Navigation = () => {
         </div>
 
         <div className="flex flex-col gap-2 px-3">
-          <button className="flex items-center justify-center w-full gap-1 py-2 font-semibold text-indigo-500 border rounded-full hover:text-black hover:border-gray-300">
+          <Button
+            variant="outline"
+            className="flex items-center justify-center w-full gap-1 py-2 font-semibold text-indigo-500 border rounded-full hover:text-black hover:border-gray-300"
+            onClick={handleDialog}
+          >
             <BoltIcon className="w-4 h-4" />
             Upgrade to Pro
-          </button>
+          </Button>
```

Comment:
Here you should directly insert the dialog logic (sample on shadcn dialogs section)
And only create a new component for the dialog content.